relayer/chain/ethereum/listener.go

```diff
@@ -117,23 +140,64 @@ func (li *Listener) pollEventsAndHeaders(
 			}
 
 			finalizedBlockNumber := gethheader.Number.Uint64() - descendantsUntilFinal
-			var events []*outbound.ContractMessage
-
+			var channelEvents []*outbound.ContractMessage
 			for _, channelContract := range li.contracts {
-				channelEvents, err := li.queryEvents(ctx, channelContract, finalizedBlockNumber, &finalizedBlockNumber)
+				contractEvents, err := li.queryChannelEvents(ctx, channelContract, finalizedBlockNumber, &finalizedBlockNumber)
 				if err != nil {
 					li.log.WithError(err).Error("Failure fetching event logs")
 				}
 
-				events = append(events, channelEvents...)
+				channelEvents = append(channelEvents, contractEvents...)
+			}
+			li.forwardChannelEvents(ctx, hcs, channelEvents)
+
+			// Query PolkadotRelayChainBridge contract's InitialVerificationSuccessful events
+			blockNumber := gethheader.Number.Uint64()
+			var events []*polkadotrelaychainbridge.ContractInitialVerificationSuccessful
+
+			contractEvents, err := li.queryLightClientEvents(ctx, blockNumber, &blockNumber)
```

Comment:
for both queryLightClientEvents and processLightClientEvents we should think about how they will work with multiple relayers running. for now, at a glance i think it looks fine in the listener, but want to think through further and also haven't looked at writer yet

Reply:
maybe we should filter out and ignore all light client events that don't correspond to our own node?

Reply:
Such that we only attempt to send the CompleteVerification tx if our relayer's Ethereum address was the sender of the CreateVerification tx? Makes sense as we're already requiring that the `msg.sender` is the same in the contract's `completeSignatureCommitment` method.

Should be fine for happy path, but later on will need to allow other nodes to submit the CompleteVerification tx after some time if the original CreateVerification tx sender doesn't complete it.

Reply:
yep. and yeh though in V1 the expectation is that other validators will do their own initial+complete tx's rather than take over. after stuff is working, we'll implement some kinda lock/stake or similar mutex to avoid redundant txs

Reply:
Added a check so we only process events from our node in https://github.com/Snowfork/polkadot-ethereum/pull/314/commits/c66da2228c36b070ca5b2db081936f342c37dc35